### ISSUE TITLE
Environnements Variables Behavior have change Beetween flowforge instance and device agent

### ISSUE THREAD

**Issue by fdplourde** [flowforge/flowforge#2408]
### Current Behavior

When I declare an environment variable with a name and value in a Flowforge instance, if I haven't declared it in a device agent's environment variables, the device agent will automatically consider the variable from the Flowforge instance and use it.

### Expected Behavior

I want the variable from the Flowforge instance and the device agent to work independently. I understand that they used to function that way before, but now I need to modify the program to accommodate this new behavior.

### Environment

- FlowForge version:v1.8.0
- Node.js version: 16.20.0
- npm version:
- Platform/OS: Digital Ocean
- Browser: Chrome

### Have you provided an initial effort estimate for this issue?

I can not provide an initial effort estimate

**Comment by MarianRaphael** [comment]
Thank you for bringing up this issue, @fdplourde 
To ensure I understand your concerns correctly, you'd like to set the environment variables for each device independently of its NR-instance in FlowForge, correct? Would you prefer this setting to apply to every single device? 
I would appreciate any additional information you can share to help us address this issue more effectively.

**Comment by fdplourde** [comment]
Yes, exactly. I'm using environment variables to limit the number of NR instances in Flowforge. The NR instances in Flowforge are only used for developing the app and debugging.

Let's say I have 100 devices, all with the same NR instance, but with the environment variable changing for each device, like "DEVICE_NO=XX." If the Flowforge NR instance has the environment variable "DEVICE_NO=99," it works well because the environment variable of the device takes precedence.

However, if I have an environment variable dedicated to the cloud instance, like "INSTANCE=CLOUD," and I don't use that variable for the device, the environment variable in the Flowforge NR instance will be used for all devices that don't have that specific environment variable. This behavior is different from before.

I'm not referring to the environment variables directly in Node-RED, but rather the ones in the instances and devices of Flowforge.

**Comment by joepavitt** [comment]
> The environment variable in the Flowforge NR instance will be used for all devices that don't have that specific environment variable. This behaviour is different from before.

Just to be sure I understand (I'm currently doing some UI work on Devices and want to ensure we consider this), you'd want the option to _not_ inherit some environment variables from the Instance, and _only_ have access to the variables defined at the Device-Level.
